File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import *

def solution(participant, completion):
    answer = ''
    c = Counter(e for e in completion)
    p = Counter(e for e in participant)
    result = (p- c)
    # print(result)
    answer = list(result.keys())[0]
    # print(answer)
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> mp;
    for(auto s:participant){
        mp[s]++;
    }
    
    for(auto s:completion){
        mp[s]--;
    }
    
    for(auto s:mp){
        if(s.second>0) return s.first;
    }
    return answer;
}